### PR TITLE
fix(wandb): allow use of sweeps

### LINKED ISSUE #1290: bug(logger): wandb fails on sweep
## 🐛 Bug

When using `wandb` sweeps for hyperparameters search, I get this error:
> wandb: ERROR Attempted to change value of key "dropout_std" from 0.030424838979365657 to 0.030424838979365654

The reason is I ran:
```python
wandb_logger.log_hyperparams(params)
```

Which I guess has some problem with floating-point numbers in high accuracy?

### PR DESCRIPTION
**Pull request by borisdayma**
# Before submitting

- [X] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
*not applicable*
- [X] Did you write any new necessary tests?  
- [X] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Fixes #1290 .

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by borisdayma** [comment]
The test has been updated but the way it's set up, we only see that the function has been called -> see [here](https://github.com/PyTorchLightning/pytorch-lightning/pull/1512/files#diff-190b7fa5d1473376da10f1de78201f0aR26)

This is due to the fact that wandb uses context and it does not work well with pytest (there was some errors with some platforms tested such as windows).

Similar behavior happens with other loggers which also use mocks and verify only that the correct function has been called.

Adding another test overwriting a previous value would only check the same function has been called (so no more than current test scope).
